src/app/context/AccountContext.tsx

```diff
@@ -93,46 +96,50 @@ export function AccountProvider({ children }: { children: React.ReactNode }) {
     setAccountBalance(balance);
   };
 
-  const fetchAccountInfo = async (options?: { accountId?: string }) => {
-    const id = options?.accountId || account?.id;
-    if (!id) return;
+  // hook: will be called each round, pass "id = null" to skip
+  const fetchAccountInfo = async () => {
```

Comment:
This doesn't seem right. SWR hooks should be used within components or other hooks.

You can use an effect when the data from the SWR hook changes, to update the account info and balance, etc.

Reply:
I refactored to call `useSWR` to the component level. This really simplifies things.